lib/neo4j/active_node/persistence.rb

```diff
@@ -128,17 +128,14 @@ def create!(*args)
         end
       end
 
-      def merge(attributes)
-        neo4j_session.query.merge(n: {self.mapped_label_names => attributes})
-          .on_create_set(n: on_create_props(attributes))
-          .on_match_set(n: on_match_props)
+      def merge(match_attributes, update_attributes = {})
+        find_or_create_query(match_attributes, update_attributes)
+          .on_match_set(n: on_match_props(update_attributes))
```

Comment:
Ahh, that's true. It assumes that you want the `match_attributes.merge update_attributes` in the event that it doesn't exist. Do you have an example of how you'd like for that to work?